### PR TITLE
Add Github Access Token clear checkbox

### PR DESCRIPTION
**Pull request by FedericoNembrini**
#### What does this PR do and why is it necessary?
Implementation of #4541
This PR adds the possibility of removing the Github Access Token once inserted.

#### How was it tested? How can it be tested by the reviewer?
Start OctoPrint, insert token, remove token.

#### What are the relevant tickets if any?
#4541

#### Screenshots (if appropriate)
image

### CODE REVIEW

**Review by foosel — CHANGES_REQUESTED** [review]
For the sake of consistency this should match the visuals of clearing an API key:

image

Implement it as an action button that triggers a settings save on the backend with the token set to undefined. That way you also don't need an additional observable just to track the state of a checkbox.

**Comment by foosel** [comment]
FYI, seen the review request, it's on my TODO list, but I'm currently fully focused on 1.8.3.

File: src/octoprint/plugins/softwareupdate/templates/snippets/githubToken.jinja2 [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
-        <input type="text" class="input-block-level" data-bind="value: config_githubToken" placeholder="{{ _('Enter a new token...')|edq }}">
-        <!-- /ko -->
+        <div class="input-append input-block-level ">
+            <!-- ko if : config_githubTokenSet -->
+            <input type="text" class="input-block-level" data-bind="value: config_githubToken" placeholder="{{ _('Token entered but hidden for privacy reasons')|edq }}">
+            <!-- /ko -->
+            <!-- ko ifnot : config_githubTokenSet -->
+            <input type="text" class="input-block-level" data-bind="value: config_githubToken" placeholder="{{ _('Enter a new token...')|edq }}">
+            <!-- /ko -->
+            <a class="btn add-on btn-danger" title="Delete API Key" data-bind="click: function() { $root.deleteGithubTokenApiKey(); }, css: {'disabled': !$root.config_githubTokenSet()}"><i class="far fa-trash-alt"></i></a>
```
Comment (cp2004) [inline src/octoprint/plugins/softwareupdate/templates/snippets/githubToken.jinja2:11]: Just for consistency 👇 

```suggestion
            <a class="btn add-on btn-danger" title="Delete GitHub Access Token" data-bind="click: function() { $root.deleteGithubTokenApiKey(); }, css: {'disabled': !$root.config_githubTokenSet()}"><i class="far fa-trash-alt"></i></a>
```

**Review by cp2004 — CHANGES_REQUESTED** [review]
Looks good to me, just one fix for consistency with the naming.

File: src/octoprint/plugins/softwareupdate/templates/snippets/githubToken.jinja2 (hunk 1 above)
Comment (FedericoNembrini) [reply src/octoprint/plugins/softwareupdate/templates/snippets/githubToken.jinja2]: Done 👍

**Review by cp2004 — APPROVED** [review]

**Comment by FedericoNembrini** [comment]
I don't know if I need to re-request review from @foosel. Just to be sure, I will.

**Comment by foosel** [comment]
Looking good now, thank you!

**Pull request merged**